src/figuresClasses.ts

```diff
@@ -1,11 +1,74 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape: Shape = 'triangle',
+    public trianglePerimeter: number = a + b + c,
+  ) {
+    const longestSide: number = Math.max(a, b, c);
+    const shortestSide: number = Math.min(a, b, c);
+    const otherSide: number = trianglePerimeter - longestSide - shortestSide;
+    const allSidesPositive: boolean =
+      longestSide >= 0 && shortestSide >= 0 && otherSide >= 0;
+
+    if (longestSide >= shortestSide + otherSide && allSidesPositive) {
+      throw new Error(`sides ${a}, ${b}, and ${c} can't form a triangle`);
```

Comment:
[CODE STYLE] - If you throw an error try to provide its detailed description in error message. The error message should explain why these sides can't form a triangle. For example, 'The sum of the lengths of any two sides of a triangle must be greater than the length of the third side.'